core/doc_data.cpp

```diff
@@ -31,10 +31,24 @@
 #include "doc_data.h"
 
 String DocData::get_default_value_string(const Variant &p_value) {
-	if (p_value.get_type() == Variant::ARRAY) {
+	const Variant::Type type = p_value.get_type();
+	if (type == Variant::ARRAY) {
 		return Variant(Array(p_value, 0, StringName(), Variant())).get_construct_string().replace("\n", " ");
-	} else if (p_value.get_type() == Variant::DICTIONARY) {
+	} else if (type == Variant::DICTIONARY) {
 		return Variant(Dictionary(p_value, 0, StringName(), Variant(), 0, StringName(), Variant())).get_construct_string().replace("\n", " ");
+	} else if (type == Variant::INT) {
+		return itos(p_value);
+	} else if (type == Variant::FLOAT) {
+		// Since some values are 32-bit internally, use 32-bit for all
+		// documentation values to avoid garbage digits at the end.
+		const String s = String::num_scientific((float)p_value);
```

Comment:
Will it matter for real_t's doubles that the values are truncated?

Reply:
This only affects the documentation, in which we need to avoid showing unnecessary digits. Widening it will result in junk being printed to the docs in a few hundred places (as opposed to the current implementation that mimics master).